tests/operator/suite.go

```diff
@@ -115,9 +115,62 @@ func LoadChecks() {
 			testMultipleSameOperators(c, &env)
 			return nil
 		}))
+
+	checksGroup.Add(checksdb.NewCheck(identifiers.GetTestIDAndLabels(identifiers.TestOperatorInstallationInTenantNamespace)).
+		WithSkipCheckFn(testhelper.GetNoOperatorsSkipFn(&env)).
+		WithCheckFn(func(c *checksdb.Check) error {
+			testOperatorInstallationInTenantNamespace(c, &env)
+			return nil
+		}))
+}
+
+/*
+Checks :
+
+ 1. Operators whose InstallTypeMode is not SingleNamespace must not be installed in the namespaces
+    specified by targetNamespace in the OperatorGroup of the operators
+
+ 2. Operators that are SingleNamespace must have CRs in only tenant namespace
+*/
+func testOperatorInstallationInTenantNamespace(check *checksdb.Check, env *provider.TestEnvironment) {
+	check.LogInfo("Starting testInstalledSingleNamespaceOperatorInTenanttNamespace")
+
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
+
+	check.LogInfo("Total operators found %d ", len(env.Operators))
+
+	for _, operator := range env.Operators {
+		check.LogInfo("Checking operator %s in namespace %s ", operator.Name, operator.Namespace)
+
+		csv := operator.Csv
+		isSingleNamespaceInstallModeSupported := hasOperatorInstallModeSingleNamespace(csv.Spec.InstallModes)
+
+		csvNamespace := csv.Namespace
+		operatorNamespace := csv.Annotations["olm.operatorNamespace"]
+		targetNamespaces := operator.TargetNamespaces
```

Comment:
Maybe I'm wrong but, to me, "tenant app namespaces" are the ones declared in certsuite_config.yaml (field [targetNamespaces](https://github.com/redhat-best-practices-for-k8s/certsuite/blob/9718894c4bac4fd05a0acc35b4b5adb17d2fac18/config/certsuite_config.yml#L1)), not the operator.TargetNamespaces (which comes from [operatorGroup.spec.targetNamespaces](https://github.com/redhat-best-practices-for-k8s/certsuite/blob/9718894c4bac4fd05a0acc35b4b5adb17d2fac18/pkg/provider/operators.go#L192)).

So isn't it good enough just checking whether the operator installation namespace (operator.Csv.Annotations["olm.operatorNamespace"]) is in the certsuite_config.yaml's "targetNamespaces" field?

Reply:
As per the requirement, the tenant dedicated **operator** namespace is mentioned by the namespace where CR is allowed to be created. In any case, we are not considering any other namespace other than the config file specified namespaces for our testing. But the check on targetNamespace is better to be done in the operator specific way, instead of going back to our configuration.

Reply:
The requirement for this test is:
```
Operators are not permitted to be installed into a tenant app namespace. Operators that are installed with the "Single Namespace" olm install mode must be installed into the tenants dedicated operator namespace. Upstream 3rd party operators will be installed Globally with "All Namespaces" olm install mode into the openshift-operators namespace.
```
In my understanding, the idea of this test is to make sure that tenants dedicate a special namespace for installing operator(s) and that it is not shared with other CNF application namespaces. So I would expect the dedicated operator namespace to be part of the certsuite_config.yaml's "targetNamespaces" filed as @greyerof  mentioned, since this hold the namespaces under test. Also, this reserved namespace should only be holding resources deployed/owned by the operator. So if non-operator pods are found in a namespace containing operator pods/resources, the test should fail.